core/utils.py

```diff
@@ -75,7 +79,7 @@ def random_perturb_dir(preprocessor, scale, categorical_prob, p1, dir, num_featu
     for feature in cat_features:
         # set the original category to -1 and a new category to +1
         if np.random.random() <= categorical_prob:
-            ohe_columns = preprocessor.get_feature_names_out([feature])
+            ohe_columns = adapter.get_feature_names_out([feature])
```

Comment:
Please consider renaming `ohe` to a more descriptive name. This goes for other non-ubiquitous abbreviations or acronyms or those which are not very apparent.

Reply:
We can revisit this when we push for open source quality. Both core/utils.py and visualize/two_d_plots.py have low-priority TODOs requesting refactor and can be ignored for now.